app_dart/lib/src/service/github_checks_service.dart

```diff
@@ -205,35 +172,4 @@ class GithubChecksService {
         throw StateError('unreachable');
     }
   }
-
-  /// Given a [headSha] and [checkSuiteId], finds the [PullRequest] that matches.
-  Future<github.PullRequest?> findMatchingPullRequest(
```

Comment:
Was this dead code?

Reply:
The only client was v1 `Scheduler`, so removing that removed the need for this. v2 `Scheduler` uses the v2 version of `GithubChecksService`, which has its own copy of this.

(There's a lot of duplication right now; this PR is removing a lot of it, and then the rest will come out in a follow-up later.)

Reply:
SGTM. I spot checked this for obvious copy paste mistakes, but I figure if any subtle bugs we'll catch in prod :)